app/assets/javascripts/modules/highlight-active-section-heading.js

```diff
@@ -1,148 +1,166 @@
-        }
+  HighlightActiveSectionHeading.prototype.checkScroll = function () {
+    if (this._hasScrolled) {
+      this._hasScrolled = false
+      var windowDimensions = this.getWindowDimensions()
+      if (windowDimensions.width <= 768) {
+        this.removeActiveItem()
+      } else {
+        this.updateActiveNavItem()
       }
     }
+  }
 
-    self.getAnchors = function () {
-      $.each(self.$anchors, function (i) {
-        var anchorID = $(this).attr('href')
-        // e.g. anchorIDs['#meeting-the-digital-service-standard', '#understand-your-users', '#research-continually']
-        anchorIDs.push(anchorID)
-      })
+  HighlightActiveSectionHeading.prototype.getWindowDimensions = function () {
+    return {
+      height: window.innerHeight,
+      width: window.innerWidth
     }
+  }
 
-    self.getHeadingPosition = function ($theID) {
-      return $theID.offset()
+  HighlightActiveSectionHeading.prototype.getAnchors = function () {
+    for (var i = 0; i < this.anchors.length; i++) {
+      var anchorID = this.anchors[i].getAttribute('href')
+      // e.g. anchorIDs['#meeting-the-digital-service-standard', '#understand-your-users', '#research-continually']
+      this.anchorIDs.push(anchorID)
     }
+  }
 
-    self.getNextHeadingPosition = function ($theNextID) {
-      return $theNextID.offset()
+  HighlightActiveSectionHeading.prototype.updateActiveNavItem = function () {
+    var windowVerticalPosition = this.getWindowPositions()
+    var footer = document.querySelector('.govuk-footer')
+    if (footer) {
+      var footerPosition = this.getFooterPosition(footer)
     }
 
-    self.getFooterPosition = function ($theID) {
-      return $theID.offset()
-    }
+    for (var i = 0; i < this.anchors.length; i++) {
+      var theID = this.anchorIDs[i]
+      var theNextID = this.anchorIDs[i + 1]
 
-    self.getDistanceBetweenHeadings = function (headingPosition, nextHeadingPosition) {
-      var distanceBetweenHeadings = (nextHeadingPosition - headingPosition)
-      return distanceBetweenHeadings
-    }
+      var $theID = document.getElementById(theID.substring(1)) // remove the # at the start
+      if (theNextID) {
+        var $theNextID = document.getElementById(theNextID.substring(1)) // remove the # at the start
+      }
 
-    self.updateActiveNavItem = function () {
-      var windowVerticalPosition = self.getWindowPositions().scrollTop
-      var footerPosition = self.getFooterPosition($('.govuk-footer'))
+      var headingPosition = this.getHeadingPosition($theID)
+      if (!headingPosition) {
+        return
+      }
 
-      $.each(self.$anchors, function (i) {
-        var theID = anchorIDs[i]
-        var theNextID = anchorIDs[i + 1]
+      headingPosition = headingPosition - 53 // fix the offset from top of page
 
-        var $theID = $(theID)
-        var $theNextID = $(theNextID)
+      if (theNextID) {
+        var nextHeadingPosition = this.getNextHeadingPosition($theNextID)
+      }
```

Comment:
Similar point on tricky variable defining where a ternary might help

Reply:
Thanks, have updated.

Reply:
Hrm, for some reason this change causes the tests to fail:

```
var nextHeadingPosition = theNextID ? this.getNextHeadingPosition($theNextID) : null
```

Not sure what's going on there. I haven't got the capacity right now to dig into this - I'd like to use this optimisation, but currently can't see how.

Reply:
@kevindew can I get your thoughts on this? I'd like to optimise this but haven't got time to investigate it properly, so for the sake of a few lines I'm happy to leave it for now.

Reply:
Just been taking a look. 

The reason it fails with the change and passes previously is because, in the previous implementation if theNextID isn't a truthy value then the previous value for nextHeadingPosition will be used. Whereas with the ternary operator it'll give nextPosition a deliberate null value.

I can't really tell if the previous implementation is desired behaviour or a bug, but I think it's simpler that I approve this since it won't be changing.

Reply:
Thank you!